### PR TITLE
test: add integration test with concurrent request load

### PR DESCRIPTION
**Pull request by fredmaggiowski**
Small PR introducing an additional integration test with a concurrent load of 100_000 requests

### CODE REVIEW

**Review by Danielecina — APPROVED** [review]

File: integration_test.go [hunk 1]
```diff
@@ -80,8 +84,174 @@ func BenchmarkStartup(b *testing.B) {
...
+		Reply(200).
+		File(oasFileName)
+
+	gock.New("http://localhost:3050/").
+		Persist().
+		Get("/allow-get").
+		Reply(200)
+	gock.New("http://localhost:3050/").
+		Persist().
+		Get("/filter-something").
```
Comment (davidebianchi) [inline integration_test.go:166]: It could be interesting assert here that a correct filter is generated?

File: integration_test.go [hunk 2]
```diff
@@ -80,8 +84,174 @@ func BenchmarkStartup(b *testing.B) {
...
+		Path           string
+		ExpectedStatus int
+	}
+	dictionary := []RequestConf{
+		{Verb: http.MethodGet, Path: "/allow-get", ExpectedStatus: http.StatusOK},
+		{Verb: http.MethodPost, Path: "/allow-get", ExpectedStatus: http.StatusForbidden},
+		{Verb: http.MethodGet, Path: "/filter-something", ExpectedStatus: http.StatusOK},
+		{Verb: http.MethodPost, Path: "/filter-something", ExpectedStatus: http.StatusNotFound},
+		{Verb: http.MethodPost, Path: "/project-data", ExpectedStatus: http.StatusOK},
+		{Verb: http.MethodGet, Path: "/project-data", ExpectedStatus: http.StatusNotFound},
```
Comment (davidebianchi) [inline integration_test.go:211]: Also here, can be interesting to assert that data is correctly projected?

**Review by davidebianchi — COMMENTED** [review]
Great! We needed a test to put the service under pressure!

File: integration_test.go (hunk 2 above)
Comment (fredmaggiowski) [reply integration_test.go:219]: done

**Comment by fredmaggiowski** [comment]
> Great! We needed a test to put the service under pressure!

I wonder if using the same test bed for a benchmark could be useful as well

**Pull request merged**